src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,49 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
-    public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+    public static String getSalaryInfo(
+            String[] names,
+            String[] data,
+            String dateFrom,
+            String dateTo) {
+
+        StringBuilder stringBuilder = new StringBuilder();
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        LocalDate from = LocalDate.parse(dateFrom, formatter);
+        LocalDate to = LocalDate.parse(dateTo, formatter);
+
+        int[] result = new int[names.length];//зберігає результат
+
+        for (String record : data) {
+
+            String[] parts = record.split(" ");
+            LocalDate workDate = LocalDate.parse(parts[0], formatter);
+            String nameOfRecord = parts[1];
+            int workHour = Integer.parseInt(parts[2]);
+            int salaryPerHour = Integer.parseInt(parts[3]);
+
+            if (workDate.isAfter(from) && workDate.isBefore(to)
+                    || workDate.isEqual(from) || workDate.isEqual(to)) {
```

Comment:
The condition for checking if `workDate` is within the range might be misleading. Ensure that the logic correctly includes both `from` and `to` dates as intended. The current condition `workDate.isAfter(from) && workDate.isBefore(to) || workDate.isEqual(from) || workDate.isEqual(to)` should work, but double-check if it aligns with the task requirements for inclusivity.